### PR TITLE
Added oled25 handwired keyboard

### PR DESCRIPTION
**Pull request by asdfire1**
## Description
Added a handwired macropad/numpad with a small OLED display.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/oled25/rules.mk [hunk 1]
```diff
@@ -0,0 +1,3 @@
+VIA_ENABLE = yes
```
Comment (zvecr) [inline keyboards/handwired/oled25/rules.mk:1]: Should only be enabled at the keymap level (also never within the default keymap). The convention would be to add a `via` keymap.
```suggestion
```

File: keyboards/handwired/oled25/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,49 @@
+
+// Copyright 2022 Antoni Malecki (@asdfire1)
```
Comment (zvecr) [inline keyboards/handwired/oled25/keymaps/default/keymap.c:2]:
```suggestion
// Copyright 2022 Antoni Malecki (@asdfire1)
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/handwired/oled25/rules.mk (hunk 1 above)
Comment (asdfire1) [reply keyboards/handwired/oled25/rules.mk]: added this change and a separate via keymap

File: keyboards/handwired/oled25/info.json [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        "rows": ["B5", "B6", "B2", "B3", "B1"]
+    },
+    "processor": "atmega32u4",
+    "url": "https://github.com/asdfire1",
+    "usb": {
+        "device_version": "1.0.0",
+        "pid": "0x3546",
+        "vid": "0xF095"
+    },
+    "layouts": {
```
Comment (zvecr) [inline keyboards/handwired/oled25/info.json:26]:
```suggestion
    "community_layouts": ["ortho_5x5"],
    "layouts": {
```

File: keyboards/handwired/oled25/readme.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
+# oled25
```
Comment (zvecr) [inline keyboards/handwired/oled25/readme.md:1]: Could you align this, and add content based on <https://github.com/qmk/qmk_firmware/blob/master/data/templates/keyboard/readme.md#bootloader>

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/handwired/oled25/readme.md (hunk 4 above)
Comment (asdfire1) [reply keyboards/handwired/oled25/readme.md:1]: I transferred my description into the template so it should be correct now. 
I'm not sure if the way I have the supported hardware and hardware availability filled out correctly for a handwired keyboard so please let me know if it needs to be changed.

File: keyboards/handwired/oled25/config.h [hunk 5]
```diff
@@ -0,0 +1,8 @@
+// Copyright 2022 Antoni Malecki (@asdfire1)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+/* Debounce reduces chatter (unintended double-presses) - set 0 if debouncing is not needed */
+#define DEBOUNCE 5
```
Comment (waffle87) [inline keyboards/handwired/oled25/config.h:7]:
```suggestion
```

File: keyboards/handwired/oled25/keymaps/default/readme.md [hunk 6]
```diff
@@ -0,0 +1 @@
+# The default keymap for oled25
```
Comment (waffle87) [inline keyboards/handwired/oled25/keymaps/default/readme.md:1]: This file should be removed.

File: keyboards/handwired/oled25/keymaps/via/readme.md [hunk 7]
```diff
@@ -0,0 +1 @@
+# The default via supported keymap for oled25
```
Comment (waffle87) [inline keyboards/handwired/oled25/keymaps/via/readme.md:1]: This file should be removed.

File: keyboards/handwired/oled25/rules.mk [hunk 8]
```diff
@@ -0,0 +1,2 @@
+OLED_ENABLE     = yes
```
Comment (waffle87) [inline keyboards/handwired/oled25/rules.mk:1]: If OLEDs are enabled at the keyboard level, functionality for them should exist at the keyboard level as well.
eg. in a `oled25.c` file using the `oled_task_kb()` function.

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/handwired/oled25/readme.md [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+
+![oled25](https://i.imgur.com/b3ODB4g.jpg)
```
Comment (drashna) [inline keyboards/handwired/oled25/readme.md:3]:
```suggestion
![oled25](https://i.imgur.com/b3ODB4gh.jpg)
```

File: keyboards/handwired/oled25/rules.mk (hunk 8 above)
Comment (waffle87) [reply keyboards/handwired/oled25/rules.mk:1]: If OLEDs are meant to be installed, functionality for them should be present at the keyboard level. There are lots of keyboards that do this — you can run `grep -r "oled_task_kb" keyboards` from within your `qmk_firmware` directory to list them.

**Pull request closed without merging**